ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/IdentityProviderConfigServiceIT.scala

```diff
@@ -0,0 +1,489 @@
+        exceptionMessageSubstring = None,
+      )
+
+      assertGrpcError(
+        t = emptyUpdateMask,
+        errorCode =
+          LedgerApiErrors.Admin.IdentityProviderConfig.InvalidUpdateIdentityProviderConfigRequest,
+        exceptionMessageSubstring = Some("update mask contains no entries"),
+      )
+
+    }
+  })
+
+  test(
+    "IdentityProviderDeleteConfigInvalidArguments",
+    "Test argument validation for IdentityProviderConfigService#DeleteIdentityProviderConfig",
+    allocate(NoParties),
+    enabled = _.userManagement.supported,
+    disabledReason = "requires user management feature",
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    def createAndCheck(
+        problem: String,
+        expectedErrorCode: ErrorCode,
+        identityProviderId: String,
+    ): Future[Unit] = {
+      for {
+        throwable <- ledger
+          .deleteIdentityProviderConfig(
+            DeleteIdentityProviderConfigRequest(identityProviderId)
+          )
+          .mustFail(context = problem)
+      } yield assertGrpcError(
+        t = throwable,
+        errorCode = expectedErrorCode,
+        exceptionMessageSubstring = None,
+      )
+    }
+
+    for {
+      _ <- createAndCheck(
+        "empty identity_provider_id",
+        LedgerApiErrors.RequestValidation.MissingField,
+        identityProviderId = "",
+      )
+      _ <- createAndCheck(
+        "invalid identity_provider_id",
+        LedgerApiErrors.RequestValidation.InvalidField,
+        identityProviderId = "!@",
+      )
+    } yield ()
+  })
+
+  test(
+    "IdentityProviderCreateConfig",
+    "Exercise CreateIdentityProviderConfig",
+    allocate(NoParties),
+    enabled = _.userManagement.supported,
+    disabledReason = "requires user management feature",
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    val identityProviderId: String = UUID.randomUUID().toString
+    val isDeactivated: Boolean = false
+    val issuer: String = UUID.randomUUID().toString
+    val jwksUrl: String = "http://daml.com/jwks.json"
+    for {
+      response1 <- ledger.createIdentityProviderConfig(
+        identityProviderId,
+        isDeactivated,
+        issuer,
+        jwksUrl,
+      )
+      response2 <- ledger.createIdentityProviderConfig(
+        isDeactivated = true
+      )
+      sameIdpIdExists <- ledger
+        .createIdentityProviderConfig(
+          identityProviderId,
+          isDeactivated,
+          UUID.randomUUID().toString,
+          jwksUrl,
+        )
+        .mustFail("Creating duplicate IDP with the same ID")
+
+      sameIssuerExists <- ledger
+        .createIdentityProviderConfig(
+          issuer = issuer
+        )
+        .mustFail("Creating duplicate IDP with the same issuer")
+
+    } yield {
+      assertIdentityProviderConfig(response1.identityProviderConfig) { config =>
+        assertEquals(config.identityProviderId, identityProviderId)
```

Comment:
Would it be more ergonomic to assert on the whole structure at once? `assertEquals(config, expectedConfig)`?

Reply:
very good point, done! thank you!